client/blocks/disconnect-jetpack/index.jsx

```diff
@@ -176,11 +176,11 @@ class DisconnectJetpack extends PureComponent {
 				);
 				this.props.recordGoogleEvent( 'Jetpack', 'Successfully Disconnected' );
 			},
-			() => {
+			( err ) => {
 				removeInfoNotice( notice.noticeId );
-				showErrorNotice(
-					translate( '%(siteName)s failed to disconnect', { args: { siteName: siteTitle } } )
-				);
+				if ( err.message ) {
+					showErrorNotice( err.message );
+				}
```

Comment:
What if `err.message` is undefined? I think it's better to fallback on the generic error message as before if there is no server-side one.

```suggestion
				if ( err.message ) {
					showErrorNotice( err.message );
				} else {
					showErrorNotice(
						translate( '%(siteName)s failed to disconnect', { args: { siteName: siteTitle } } )
					);
				}
```

Reply:
@manzoorwanijk Updated 👍